File: 003CreatedAndUsedFetchRequest_OSX_ObjC/003CreatedAndUsedFetchRequest_OSX_ObjC/ViewController.m
//
//  ViewController.m
//  003CreatedAndUsedFetchRequest_OSX_ObjC
//
//  Created by Henry AT on 6/19/15.
//  Copyright (c) 2015 Apps4s. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  // Do any additional setup after loading the view.
}

- (void)setRepresentedObject:(id)representedObject {
  [super setRepresentedObject:representedObject];

  // Update the view, if already loaded.
}

- (IBAction)btnCreatedManagedObject:(id)sender {
  
  //  AppDelegate *appDelegate = (AppDelegate *)[[NSApplication sharedApplication] delegate];
  //  NSManagedObjectContext *context = appDelegate.managedObjectContext;
  NSManagedObjectContext *context = ((AppDelegate *)[[NSApplication sharedApplication] delegate]).managedObjectContext;
  
  NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
  [dateFormatter setDateFormat:@"dd MMM yyyy"];
  
  // First Course Object
  Course *first = (Course *)[NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
  first.title = @"Core Data for iOS and OS X";
  first.author = @"Henry AT";
  first.releaseDate = [dateFormatter dateFromString:@"19 Jun 2015"];
  [first simpleMethod];
  
  // Second Course Object
  Course *second = (Course *)[NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
  second.title = @"iOS SDK";
  second.author = @"Henry AT";
  second.releaseDate = [dateFormatter dateFromString:@"19 Jun 2015"];
  [second simpleMethod];
  
  // Third Course Object
  Course *third = (Course *)[NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
  third.title = @"Java Essential";
  third.author = @"Henry AT";
  third.releaseDate = [dateFormatter dateFromString:@"19 Jun 2015"];
  [third simpleMethod];
  
  // Fourth Course Object
  Course *fourth = (Course *)[NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
  fourth.title = @"C/C++ Essential";
  fourth.author = @"Henry AT";
  fourth.releaseDate = [dateFormatter dateFromString:@"19 Jun 2015"];
  [fourth simpleMethod];
  
  // Fifth Course Object
  Course *fifth = (Course *)[NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
  fifth.title = @"Cocoa Essential";
  fifth.author = @"Henry AT";
  fifth.releaseDate = [dateFormatter dateFromString:@"19 Jun 2015"];
  [fifth simpleMethod];
  
  // Save the Managed Object Context
  
  NSError *error = nil;
  
  if (![context save:&error]) {
    NSLog(@"An Error! %@",error);
  } else{
    NSLog(@"Todo se grabo OK..");
  }
  
}

- (IBAction)btnFetchManagedObject:(id)sender {

  NSManagedObjectContext *context = ((AppDelegate *)[[NSApplication sharedApplication] delegate]).managedObjectContext;
  
  NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
  NSEntityDescription *entity = [NSEntityDescription entityForName:@"Course" inManagedObjectContext:context];
  [fetchRequest setEntity:entity];
  
  
  NSError *error = nil;
  NSArray *fetchedObjects = [context executeFetchRequest:fetchRequest error:&error];
  if (fetchedObjects == nil) {
    NSLog(@"Problme! %@",error);
  }
  
  for (Course * c in fetchedObjects) {
    NSLog(@"Course: %@ by author: %@",c.title, c.author);
  }
  
}












@end

File: 003CreatedAndUsedFetchRequest_OSX_Swift/003CreatedAndUsedFetchRequest_OSX_Swift/ViewController.swift
//
//  ViewController.swift
//  003CreatedAndUsedFetchRequest_OSX_Swift
//
//  Created by Henry AT on 6/19/15.
//  Copyright (c) 2015 Apps4s. All rights reserved.
//

import Cocoa
import CoreData

class ViewController: NSViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    // Do any additional setup after loading the view.
  }

  override var representedObject: AnyObject? {
    didSet {
    // Update the view, if already loaded.
    }
  }

  
  @IBAction func btnCreatedManagedObject(sender: AnyObject) {
    
//    let appDelegate = NSApplication.sharedApplication().delegate as! AppDelegate
//    var context: NSManagedObjectContext? = appDelegate.managedObjectContext
    var context: NSManagedObjectContext? = (NSApplication.sharedApplication().delegate as! AppDelegate).managedObjectContext
    
    
    let formatter = NSDateFormatter()
    //format yyyy-MM-dd like 2015-02-23
    formatter.dateFormat = "dd MMM yyyy";
    
    //format mm-dd-yy like 02-23-15
//    formatter.dateStyle = .ShortStyle;
    //format feb 02, 2015
//    formatter.dateStyle = .MediumStyle;
    
    
// First Course Object
    let first = NSEntityDescription.insertNewObjectForEntityForName("Course", inManagedObjectContext: context!) as! Course
    first.title = "Cocoa Essential"
    first.author = "Henry"
    first.releaseDate = NSDate()
    first.simpleMethod()
    
    // Second Course Object
    let second = NSEntityDescription.insertNewObjectForEntityForName("Course", inManagedObjectContext: context!) as! Course
    second.title = "Curso de CoreData for OS X en Swift"
    second.author = "Henry"
    second.releaseDate = NSDate()
    second.simpleMethod()
    
    // Third Course Object
    let third = NSEntityDescription.insertNewObjectForEntityForName("Course", inManagedObjectContext: context!) as! Course
    third.title = "iOS SDK"
    third.author = "Henry"
    third.releaseDate = NSDate()
    third.simpleMethod()
    
    // Fourth Course Object
    let fourth = NSEntityDescription.insertNewObjectForEntityForName("Course", inManagedObjectContext: context!) as! Course
    fourth.title = "Java Essential"
    fourth.author = "Henry"
    fourth.releaseDate = NSDate()
    fourth.simpleMethod()
    
    // Fifth Course Object
    let fifth = NSEntityDescription.insertNewObjectForEntityForName("Course", inManagedObjectContext: context!) as! Course
    fifth.title = "C/C++ Essential"
    fifth.author = "Henry"
    fifth.releaseDate = NSDate()
    fifth.simpleMethod()
    
  // Save the Managed Object Context
    
    // Ahora grabaremos, para no tener la necesidad de salir del Simulador
    var error: NSError?
    
    if((context?.save(&error)) != nil){
      println("Todo se grabo OK..")
    }else{
      println("An Error! \(error!.localizedDescription)")
    }
    
    
    
  }
  
  @IBAction func btnFetchManagedObject(sender: AnyObject) {
    
    
    var context: NSManagedObjectContext? = (NSApplication.sharedApplication().delegate as! AppDelegate).managedObjectContext
    
    
    var fetchRequest: NSFetchRequest = NSFetchRequest(entityName: "Course")
    
    var error: NSError?
    var fetchedObjects = context!.executeFetchRequest(fetchRequest, error:&error)
    if(fetchedObjects == nil){
      println("Problem! \(error)")
    }
    
    for resultItem in fetchedObjects! {
      var courseItem = resultItem as! Course
      println("Course: \(courseItem.title) by author: \(courseItem.author)")
    }

    
    
   
    
  }

  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
}

